lib/chef/provider/execute.rb

```diff
@@ -58,7 +58,16 @@ def action_run
         end
 
         converge_by("execute #{description}") do
-          result = shell_out!(command, opts)
+          begin
+            shell_out!(command, opts)
+          rescue Mixlib::ShellOut::ShellCommandFailed => e
+            if sensitive?
+              raise Mixlib::ShellOut::ShellCommandFailed,
+                "Command execution failed. STDOUT/STDERR suppressed for sensitive resource"
+            else
+              raise e
```

Comment:
I think we just want `raise` instead of `raise e` to preserve the original stack trace.